lib/_http_parser.js

```diff
@@ -0,0 +1,890 @@
+/*
+Misc differences with joyent/http-parser:
+  * Folding whitespace behavior is conformant with RFC 7230:
+
+      "A user agent that receives an obs-fold in a response message that is
+       not within a message/http container MUST replace each received
+       obs-fold with one or more SP octets prior to interpreting the field
+       value."
+
+    It should also be noted that RFC 7230 now deprecates line folding for HTTP
+    parsing, FWIW. This parser replaces folds with a single SP octet.
+
+  * Optional whitespace is removed before interpreting a header field value, as
+    suggested by RFC 7230:
+
+      "A field value might be preceded and/or followed by optional
+       whitespace (OWS); a single SP preceding the field-value is preferred
+       for consistent readability by humans.  The field value does not
+       include any leading or trailing whitespace: OWS occurring before the
+       first non-whitespace octet of the field value or after the last
+       non-whitespace octet of the field value ought to be excluded by
+       parsers when extracting the field value from a header field."
+
+    joyent/http-parser keeps trailing whitespace. This parser keeps neither
+    preceding nor trailing whitespace.
+
+  * Enforces CRLF for line endings instead of additionally allowing just LF.
+
+  * Does not allow spaces (which are invalid) in header field names.
+
+  * Smaller maximum chunk/content length (2^53-1 vs 2^64-2). Obviously it's
+    not *impossible* to handle a full 64-bit length, but it would mean adding
+    some kind of "big integer" library for lengths > 2^53-1.
+
+  * No special handling for `Proxy-Connection` header. The reason for this is
+    that `Proxy-Connection` was an experimental header for HTTP 1.0 user agents
+    that ended up being a bad idea because of the confusion it can bring. You
+    can read a bit more about this in
+    [RFC 7230 A.1.2](https://tools.ietf.org/html/rfc7230#page-79)
+*/
+
+var inspect = require('util').inspect;
+
+var CR = 13;
+var LF = 10;
```

Comment:
I wonder if it might be worthwhile to rebase on top of `next` to see how the (imminent) V8 upgrades affect perf before changing code to suit the current V8 version.